### ISSUE TITLE
Obsidian loading is incredibly slow when plugin is enabled, even with no events

### ISSUE THREAD

**Issue by asmagen** [davish/obsidian-full-calendar#329]
Obsidian loading is incredibly slow when plugin is enabled, even with no events

**Comment by davish** [comment]
I believe this is a duplicate of #324, but will need more information from you in order to confirm that. Can you send a screenshot or copy/paste the contents of the JavaScript console while Obsidian is loading?

**Comment by asmagen** [comment]
It crashes on iOS while loading -- how do I get any console info in this case?

**Comment by davish** [comment]
does this also happen on desktop?

**Comment by asmagen** [comment]
No

**Comment by Mauwzola** [comment]
I've been experiencing similar problems for a few weeks now. With about 200 events and 3500 total notes in my Vault, on my iPhone, Obsidian gets hung up halfway through the indexing function and, after a restart or two, gets hung with a white screen displayed. When I disable Full Calendar, it works fine. On the desktop, at startup, the "Loading cache..." phase takes a few minutes with Full Calendar enabled, but it takes no time at all when Full Calendar is disabled.

**Comment by asmagen** [comment]
Yes it has an issue somehow with either a lot of events, or potentially just when using multiple calendar types (daily notes + full notes), or potentially when the daily notes header has something other than the calendar event info (mixed with other non calendar rows)
________________________________

**Comment by Mauwzola** [comment]
I'm wondering if it might be related to recurring events? I don't use Full Calendar to add new events as I employ a more elaborate template. As a result, most of my event notes contain blank startRecur, endRecur, and daysOfWeek fields. Haven't looked into this, but I'm wondering if perhaps the indexing function is overthinking these entries?

**Comment by mehlsson** [comment]
I had the same and I don't think it's related to recurring events. I had 5 full calendars with I think around 500 non-recurring events, and that was too much for it. Here is another reference to the same error:

https://www.reddit.com/r/ObsidianMD/comments/1465qsv/is_anyone_having_problem_using_full_calendar/

For now I switched to having separate calendars for iOS, that only have the last week, and archive calendars that hold the rest. The issue is annoying enough that I'll maybe look into it myself in the future.
